packages/relay-compiler/bin/RelayCompilerBin.js

```diff
@@ -122,6 +128,9 @@ function getRelayFileWriter(baseDir: string) {
 function getSchema(schemaPath: string): GraphQLSchema {
   try {
     let source = fs.readFileSync(schemaPath, 'utf8');
+    if (path.extname(schemaPath) === '.json') {
+      source = printSchema(buildClientSchema(JSON.parse(source).data));
+    }
```

Comment:
@josephsavona could you give me some pointers on how to incorpore those directives in the GraphQLSchema. I am not familiar with the inner working of graphql-js, Thanks.